src/hub/HubRegistry.sol

```diff
@@ -76,18 +78,27 @@ contract HubRegistry is Auth, IHubRegistry {
     }
 
     /// @inheritdoc IHubRegistry
-    function updateCurrency(PoolId poolId, AssetId currency_) external auth {
-        require(exists(poolId), NonExistingPool(poolId));
+    function updateCurrency(PoolId poolId_, AssetId currency_) external auth {
+        require(exists(poolId_), NonExistingPool(poolId_));
         require(!currency_.isNull(), EmptyCurrency());
 
-        currency[poolId] = currency_;
+        currency[poolId_] = currency_;
 
-        emit UpdateCurrency(poolId, currency_);
+        emit UpdateCurrency(poolId_, currency_);
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // PermissViewionless methods
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IHubRegistry
+    function poolId(uint16 centrifugeId, uint48 postfix) public pure returns (PoolId poolId_) {
```

Comment:
CR

The `HubRegistry` MUST have a storage for the `centrifugeId`

Reply:
Why @mustermeiszer ?

Reply:
I think it can just be looked up in the other contract

Reply:
That is essential for a correctly formed poolId. Otherwise, the pool is on ChainA but has id from ChainB

Reply:
Of course I understand why it needs the right chain ID. I am asking, why does it need to be in the `HubRegistry` vs in the `MessageDispatcher` where it already is now?

Anyways if you feel strongly I can add it here.

Reply:
I just think you should not be able to create a pool in the first place. Or am I missing the check during pool creation?

Reply:
That would be a pool that is saying I am on Chain B while it is on Ch.ain A

Reply:
There's a check in the `Hub`